xapian-core/weight/tfidfweight.cc

```diff
@@ -223,6 +234,8 @@ TfIdfWeight::get_idfn(char c) const
     if (c != 'n') termfreq = get_termfreq();
     double N = 1.0;
     if (c != 'n' && c != 'f') N = get_collection_size();
+    Xapian::termcount collfreq = 1;
+    if (c == 'E' || c == 'G' || c == 'l') collfreq = get_collection_freq();
```

Comment:
We're about to switch on `c`, so it seems cleaner to call `get_collection_freq()` directly in the cases that need it, rather than effectively have a pre-dispatch on `c`.

(I realise this is an existing pattern in this code, but I don't think it's a good pattern to extend further)